### PR TITLE
(un)install scripts and mac support

### PR DESCRIPTION
**Pull request by thedavidmeister**
- CI working for mac
- removes all libsodium from core (because libsodium is an eternal pain to setup for mac)
- use ring for random bytes
- scripts for (un)install for binaries and report on current state as `hc-doctor` to help troubleshoot

### CODE REVIEW

**Review by maackle — CHANGES_REQUESTED** [review]
@thedavidmeister I dunno about this. Can't we just have an `hc-install` command for this? My concern is that there are already multiple binaries to be built (holochain, dna-util, and now lair), and you can't anticipate which one the user came to install. It's not really fair to make them wait for all of them to build (in release mode) before they can do anything.

**Comment by maackle** [comment]
Especially if all they care about is debug builds, then it's just cruelty

**Comment by thedavidmeister** [comment]
@maackle not sure, there's currently no way to get a `holochain` binary in the nix shell other than running `cargo install` and if we do nothing it points to the redux version, i feel this warrants discussion

install incrementally compiles so it is pretty fast after the first run

**Comment by maackle** [comment]
We discussed and it led to the decision to soon have a hard cutoff on an upcoming holonix release that no longer includes the old holochain binary

File: default.nix [hunk 1]
```diff
@@ -62,6 +63,44 @@ with holonix.pkgs;
...
+
+   ++ ([(
+    holonix.pkgs.writeShellScriptBin "hc-uninstall" ''
+     rm $CARGO_INSTALL_ROOT/bin/holochain
+     echo 'holochain uninstalled!'
+     echo
+
+     hc-doctor
+    ''
+   )])
```
Comment (maackle) [inline default.nix:84]: @thedavidmeister people need to build `dna-util` as well. And maybe lair? So, should these be namespaced, `hc-install-holochain` etc?

File: default.nix [hunk 2]
```diff
@@ -65,21 +65,51 @@ with holonix.pkgs;
...
+    holonix.pkgs.writeShellScriptBin "hc-install-dna-util" ''
+     cargo install --path crates/dna_util
+     echo 'dna util installed!'
+     echo
+    ''
+   )])
+
+   ++ ([(
+    holonix.pkgs.writeShellScriptBin "hc-uninstall-dna-util" ''
+     rm $CARGO_INSTALL_ROOT/bin/dna-util
```
Comment (neonphog) [inline default.nix:110]: Why not `cargo uninstall` if nothing else just for symmetry with the install command?

Same with `hc-uninstall-holochain` above.

**Review by maackle — APPROVED** [review]
This set of commands looks good, but I agree with neonphog that we might as well use `cargo uninstall` instead of `rm`

**Pull request merged**